docs/relationship_synthesis.md

```diff
@@ -0,0 +1,297 @@
+| Kubernetes Integration | Uses data generated by the New Relic Kubernetes Integration.      |
+| Pixie                  | Uses data generated by the New Relic Pixie Integration.           |
+| Prometheus             | Uses data generated by the New Relic Prometheus Integration.      |
+| OpenTelemetry          | Uses data generated by the New Relic OpenTelemetry Integration.   |
+| OnHost Integration     | Uses data generated by the New Relic OnHost Integration.          |
+
+One rule can be applied to multiple sources, as shown in the example. These origins are used by our internal services 
+to filter which rules to evaluate and which ones to ignore. 
+The mapping of origins to sources allows for performance improvements when evaluating rules via the matching system.
+
+## Conditions
+
+`Conditions` are used to determine if a rule should be applied to a given data point. 
+The proposal for `conditions` is an iterative improvement over the existing rules format to allow for 
+future changes without major breaking updates.
+
+The following example demonstrates various capabilities supported by conditions:
+
+```yaml
+conditions:
+  - attribute: eventType
+    anyOf: ["NginxSample", "K8sPodSample", "K8sContainerSample"]
+  - attribute: host.guid
+    present: yes
+  - attribute: metricName
+    startsWith: "DurationByCaller/"
+    caseSensitive: true
+  - attribute: metricName
+    regex: ".*\.rds\.amazonaws\.com.*"
+```
+
+The `attribute` field specifies the name of the attribute in the data point, and the operation to perform is defined using matchers 
+such as `anyOf`, `present`, `startsWith`, and `regex`. 
+Only one matcher should be present in each condition, and matching is case-insensitive by default unless 
+specified otherwise with `caseSensitive: true`. 
+Multiple conditions must all match for the rule to apply; there is no support for `OR` conditions.
+
+## Relationship
+
+The `relationship` section defines aspects related to the relationship we want to synthesize.
+
+### Expires
+
+The `expires` field allows configuring the duration for which the relationship should exist if it is not reported within that timeframe. 
+By default, this field is set to ***75 minutes*** (the default for the relationship platform). 
+The value for expires follows the `ISO-8601` format and must be between ***10 minutes*** and ***72 hours*** (inclusive), which is the time interval 
+currently supported by the platform.
+
+Explicitly providing the expires field is recommended for clarity, even though the default is 75 minutes.
+
+### RelationshipType
+
+The `relationshipType` field specifies the type of relationship to create. Only predefined values from the 
+supported relationship types list are accepted.
+
+The full list of supported relationship type options (as of July 2023) is the following:
+
+| **Relationship type** | **Description**                                                                                                                |
```

Comment:
The full list contains the content of `<internal-repo>/entity-platform/entity_platform_models/blob/master/entity-platform-models-relationships/src/main/java/com/newrelic/entityplatform/relationships/models/RelationshipType.java#L9`

I would like to bring up some questions before we go public w/ this one, to be considered from an API and storage perspectives:

- Would we like to keep maintaining it as an internal enum after it goes public? 

From an API point-of-view, maybe it is easier to maintain and explain something that is visible on the public repo.

- Do we have any concerns regarding scalability and graph sharding due to the explosion of relationship types?

If we make it "unlimited" (a maximum sized string) or open to public creation (a public enum here on this repo), it would be simpler to manage but may have storage layer implications. Have we considered those? What is our internal perspective on a possible explosion of relationship types?

Reply:
cc @naxhh @gvassallo

Reply:
we discussed offline, and since the enum needs to be fixed due to the GraphQL API, decided to just reflect the enum on the documentation table (and apply validations to the repository on PRs).